funnel/views/commentvote.py

```diff
@@ -46,19 +50,98 @@ def votedown(self):
+                                send_mail_info.append({'to': self.obj.user.email or self.obj.email,
+                                    'subject': u"{project} Funnel: {proposal}".format(project=self.obj.project.title, proposal=self.obj.title),
+                                    'template': 'proposal_comment_reply_email.md'})
+                        else:  # send mail to parent comment owner & proposal owner
+                            if not parent.user == current_auth.user:
+                                send_mail_info.append({'to': parent.user.email,
+                                    'subject': u"{project} Funnel: {proposal}".format(project=self.obj.project.title, proposal=self.obj.title),
+                                    'template': 'proposal_comment_to_proposer_email.md'})
+                            if not self.obj.user == current_auth.user:
+                                send_mail_info.append({'to': self.obj.user.email or self.obj.email,
+                                    'subject': u"{project} Funnel: {proposal}".format(project=self.obj.project.title, proposal=self.obj.title),
+                                    'template': 'proposal_comment_email.md'})
+
+                    if parent and parent.commentset == self.obj.commentset:
+                        comment.parent = parent
+                else:  # for top level comment
+                    if not self.obj.user == current_auth.user:
+                        send_mail_info.append({'to': self.obj.user.email or self.obj.email,
+                            'subject': u"{project} Funnel: {proposal}".format(project=self.obj.project.title, proposal=self.obj.title),
+                            'template': 'proposal_comment_email.md'})
+                self.obj.commentset.count += 1
+                comment.voteset.vote(current_auth.user)  # Vote for your own comment
+                db.session.add(comment)
+                flash(_("Your comment has been posted"), 'info')
+            db.session.commit()
+            to_redirect = self.obj.url_for(_external=True)
+            for item in send_mail_info:
+                email_body = render_template(item.pop('template'), proposal=self.obj, comment=comment, link=to_redirect)
+                if item.get('to'):
+                    # Sender is set to None to prevent revealing email.
+                    send_mail(sender=None, body=email_body, **item)
+        # Redirect despite this being the same page because HTTP 303 is required to not break
+        # the browser Back button
+        return redirect(to_redirect, code=303)
+
+    @route('comments/delete', methods=['POST'])
+    @lastuser.requires_login
+    @requires_permission('new_comment')
+    @requestargs('comment_id')
+    def delete_comment(self, comment_id):
+        comment = Comment.query.filter_by(id=comment_id).first_or_404()
+        if not comment.current_permissions.delete_comment:
```

Comment:
IIRC, this will break because it will only grant delete permission to the owner of a comment, whereas previously even admins had the right to delete comments. You have to use `current_auth.permissions.delete_comment` here _and_ change `after_loader` to load permissions using inheritance (`comment.permissions(current_auth.user, proposal.permissions(current_auth.user))`). The `after_loader` bit should be fixed in Coaster with the `MultiModelView` issue (ticket is pending to be filed there).

Reply:
Wait, why is the comment not being loaded in `loader`? All the background processing goes for a toss when the loading moves into the view.

Reply:
@jace this is not a CommentView. It's a ProposalView. The frontend code makes it difficult to POST to the `comments/<id>/delete` URL. The handling is very weird there. Hence the request is being made to `<proposal_id>/comments/delete` with the comment_id as POST payload.